src/EntityComponentManager_TEST.cc

```diff
@@ -437,6 +437,31 @@ TEST_P(EntityComponentManagerFixture, EntitiesAndComponents)
   EXPECT_FALSE(manager.EntityHasComponentType(entity, DoubleComponent::typeId));
   EXPECT_FALSE(manager.EntityHasComponentType(entity2, IntComponent::typeId));
 
+  // Query non-existing component, the default value is default-constructed
+  BoolComponent* boolComp = manager.ComponentDefault<BoolComponent>(entity);
```

Comment:
Style nit, also applies to others below.

```suggestion
  BoolComponent *boolComp = manager.ComponentDefault<BoolComponent>(entity);
```